sound/hda/intel-dsp-config.c

```diff
@@ -306,6 +310,21 @@ static const struct config_entry config_table[] = {
 		.flags = FLAG_SOF | FLAG_SOF_ONLY_IF_DMIC_OR_SOUNDWIRE,
 		.device = 0xa0c8,
 	},
+	/* TigerLake-H */
+	{
+		.flags = FLAG_SOF,
+		.device = 0x43c8,
+		.dmi_table = (const struct dmi_system_id []) {
+			{
+				.ident = "Dell",
+				.matches = {
+					DMI_MATCH(DMI_SYS_VENDOR, "Dell Inc"),
+				}
+			},
+			{}
+		},
+		.cpu_id_check = soc_intel_is_rkl,
```

Comment:
all the `soc_intel_is_*()` functions are declared as `inline`...

Reply:
I thought about this but it seems to compile so I am not sure what the compiler does with this.

Reply:
`inline` is a _suggestion_ to avoid a function call, it's not a suggestion not to create the function itself. Those are two different things.

While I trust the kernel to go and twist compilers' arms, it's very hard to make any _functional_ difference with `inline`.

Granted, gcc tries to be "smart":
 https://gcc.gnu.org/onlinedocs/gcc/Inline.html
> When a function is both inline and static, if all calls to the function are integrated into the caller, and the function’s address is never used, then the function’s own assembler code is never referenced. In this case, GCC does not actually output assembler code for the function, unless you specify the option -fkeep-inline-functions.

But:

>  The function must also be compiled as usual if the program refers to its address, because that cannot be inlined.

Reply:
thanks Professor!